### PR TITLE
Enable cross-compilation in Dockerfile

### PR DESCRIPTION
**Pull request by chewong**
**What this PR does / why we need it**:

Enable cross-compilation in Dockerfile. We need to add `ARG TARGETOS TARGETARCH` for them to be populated by buildx and we need to add `--platform=$TARGETPLATFORM` in `FROM` statements to specify which platform we use to pull those images

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -18,12 +18,13 @@ FROM --platform=$BUILDPLATFORM $golang_image AS builder
 WORKDIR /go/src/github.com/kubernetes-sigs/aws-iam-authenticator
 COPY . .
 RUN go mod download
+ARG TARGETOS TARGETARCH
```
Comment (nckturner) [inline Dockerfile:21]: Can we get a makefile target similar to [build-all-bins](https://github.com/kubernetes-sigs/aws-iam-authenticator/blob/master/Makefile#L81) for building a set of supported targets?

Comment (chewong) [reply Dockerfile:21]: Done

**Pull request merged**